classes/cliTool/ConvertLogFile.php

```diff
@@ -0,0 +1,1012 @@
+                continue;
+            }
+
+            $newEntry['time'] = $entryData['date'];
+
+            $ip = $entryData['ip'];
+            $ipNotHashed = preg_match('/^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$/', $ip);
+            // shell IPv6 be considered ?
+            if ($ipNotHashed === 1) {
+                $saltFileName = StatisticsHelper::getSaltFileName();
+                $salt = trim(file_get_contents($saltFileName));
+                $hashedIp = StatisticsHelper::hashIp($ip, $salt);
+                $newEntry['ip'] = $hashedIp;
+            } else {
+                // check if it is a string(64) i.e. sha256 ?
+                $newEntry['ip'] = $ip;
+            }
+
+            $newEntry['userAgent'] = $entryData['userAgent'];
+            $newEntry['canonicalUrl'] = $entryData['url'];
+
+            [$assocType, $contextPaths, $page, $op, $args] = $this->getUrlMatches($entryData['url'], $lineNumber);
+            if ($assocType && $contextPaths && $page && $op) {
+                $foundContextPath = current($contextPaths);
+                if (!array_key_exists($foundContextPath, $this->contextsByPath)) {
+                    echo "Context with the path {$foundContextPath} does not exist.\n";
+                    continue;
+                }
+                $context = $this->contextsByPath[$foundContextPath];
+                $newEntry['contextId'] = $context->getId();
+
+                $this->setAssoc($assocType, $op, $args, $newEntry);
+                if (!array_key_exists('assocType', $newEntry)) {
+                    if (!$this->isApacheAccessLogFile()) {
+                        echo "The URL {$entryData['url']} in the line number {$lineNumber} was not considered.\n";
+                    }
+                    continue;
+                }
+            } else {
+                continue;
+            }
+
+            // Geo data
+            $country = $region = $city = null;
+            if ($ipNotHashed === 1) {
+                $statisticsHelper = new StatisticsHelper();
+                $site = Application::get()->getRequest()->getSite();
+                [$country, $region, $city] = $statisticsHelper->getGeoData($site, $context, $ip, $hashedIp, false);
+            }
+            $newEntry['country'] = $country;
+            $newEntry['region'] = $region;
+            $newEntry['city'] = $city;
+
+            // institutions IDs
+            $institutionIds = [];
+            if ($ipNotHashed === 1 && $context->isInstitutionStatsEnabled($site)) {
+                $institutionIds = $statisticsHelper->getInstitutionIds($context->getId(), $ip, $hashedIp, false);
+            }
+            $newEntry['institutionIds'] = $institutionIds;
+
+            $newEntry['version'] = Registry::get('appVersion');
+
+            // write to a new file
+            $newLogEntry = json_encode($newEntry) . PHP_EOL;
+            file_put_contents($newFilePath, $newLogEntry, FILE_APPEND);
```

Comment:
I suspect it's probably not too bad, but this would probably perform better if you had another file handle open from `fopen` rather than calling `file_put_contents` many times.